File: push/src/main/java/com/xiaomi/push/log/LogUploader.java
package com.xiaomi.push.log;

import android.content.Context;
import android.content.SharedPreferences;
import com.xiaomi.channel.commonutils.file.SDCardUtils;
import com.xiaomi.channel.commonutils.logger.MyLog;
import com.xiaomi.channel.commonutils.misc.SerializedAsyncTaskProcessor;
import com.xiaomi.channel.commonutils.network.Network;
import com.xiaomi.miui.pushads.sdk.NotifyAdsDef;
import com.xiaomi.push.service.PushConstants;
import com.xiaomi.push.service.ServiceConfig;
import com.xiaomi.smack.util.TaskExecutor;
import java.io.File;
import java.io.IOException;
import java.util.Date;
import java.util.HashMap;
import java.util.Map;
import java.util.concurrent.ConcurrentLinkedQueue;
import org.json.JSONException;
import org.json.JSONObject;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/push/log/LogUploader.class */
public class LogUploader {
    private static final int EXPIRE_TIME = 172800000;
    private static final int MAX_PENDING_TASKS = 6;
    private static final int MAX_TIMES_PER_DAY = 10;
    private static final String PREF_KEY_REQUEST = "log.requst";
    private static final String PREF_NAME = "log.timestamp";
    private static final String ZIPPED_LOG_PATH = "/.logcache";
    private static volatile LogUploader sInstance = null;
    private Context mContext;
    private final ConcurrentLinkedQueue<Task> mTasks;

    /* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/push/log/LogUploader$CleanUpTask.class */
    class CleanUpTask extends Task {
        CleanUpTask() {
            super();
        }

        @Override // com.xiaomi.push.log.LogUploader.Task, com.xiaomi.channel.commonutils.misc.SerializedAsyncTaskProcessor.SerializedAsyncTask
        public void process() {
            LogUploader.this.cleanUp();
        }
    }

    /* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/push/log/LogUploader$Task.class */
    class Task extends SerializedAsyncTaskProcessor.SerializedAsyncTask {
        long timestamp = System.currentTimeMillis();

        Task() {
        }

        public boolean canExcuteNow() {
            return true;
        }

        final boolean isExpired() {
            return System.currentTimeMillis() - this.timestamp > 172800000;
        }

        @Override // com.xiaomi.channel.commonutils.misc.SerializedAsyncTaskProcessor.SerializedAsyncTask
        public void process() {
        }
    }

    /* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/push/log/LogUploader$UploadTask.class */
    class UploadTask extends Task {
        File file;
        boolean force;
        int retryNum;
        String token;
        boolean uploaded;
        String url;

        UploadTask(String str, String str2, File file, boolean z) {
            super();
            this.url = str;
            this.token = str2;
            this.file = file;
            this.force = z;
        }

        private boolean checkLimit() throws JSONException {
            int i;
            SharedPreferences sharedPreferences = LogUploader.this.mContext.getSharedPreferences(LogUploader.PREF_NAME, 0);
            String string = sharedPreferences.getString(LogUploader.PREF_KEY_REQUEST, "");
            int i2 = 0;
            long jCurrentTimeMillis = System.currentTimeMillis();
            try {
                JSONObject jSONObject = new JSONObject(string);
                long j = jSONObject.getLong(NotifyAdsDef.JSON_TAG_ACTIONTIME);
                jCurrentTimeMillis = j;
                i2 = jSONObject.getInt("times");
                jCurrentTimeMillis = j;
            } catch (JSONException e) {
            }
            if (System.currentTimeMillis() - jCurrentTimeMillis < 86400000) {
                i = i2;
                if (i2 > 10) {
                    return false;
                }
            } else {
                jCurrentTimeMillis = System.currentTimeMillis();
                i = 0;
            }
            JSONObject jSONObject2 = new JSONObject();
            try {
                jSONObject2.put(NotifyAdsDef.JSON_TAG_ACTIONTIME, jCurrentTimeMillis);
                jSONObject2.put("times", i + 1);
                sharedPreferences.edit().putString(LogUploader.PREF_KEY_REQUEST, jSONObject2.toString()).commit();
                return true;
            } catch (JSONException e2) {
                MyLog.v("JSONException on put " + e2.getMessage());
                return true;
            }
        }

        @Override // com.xiaomi.push.log.LogUploader.Task
        public boolean canExcuteNow() {
            return Network.isWIFIConnected(LogUploader.this.mContext) || (this.force && Network.hasNetwork(LogUploader.this.mContext));
        }

        @Override // com.xiaomi.channel.commonutils.misc.SerializedAsyncTaskProcessor.SerializedAsyncTask
        public void postProcess() {
            if (!this.uploaded) {
                int i = this.retryNum + 1;
                this.retryNum = i;
                if (i < 3) {
                    LogUploader.this.mTasks.add(this);
                }
            }
            if (this.uploaded || this.retryNum >= 3) {
                this.file.delete();
            }
            LogUploader.this.uploadIfNeed((1 << this.retryNum) * 1000);
        }

        @Override // com.xiaomi.push.log.LogUploader.Task, com.xiaomi.channel.commonutils.misc.SerializedAsyncTaskProcessor.SerializedAsyncTask
        public void process() {
            try {
                if (checkLimit()) {
                    Map<String, String> map = new HashMap<>();
                    map.put("uid", ServiceConfig.getDeviceUUID());
                    map.put("token", this.token);
                    map.put("net", Network.getActiveConnPoint(LogUploader.this.mContext));
                    Network.uploadFile(this.url, map, this.file, PushConstants.UPLOAD_FILE_POST_KEY);
                }
                this.uploaded = true;
            } catch (Exception e) {
            }
        }
    }

    private LogUploader(Context context) {
        ConcurrentLinkedQueue<Task> concurrentLinkedQueue = new ConcurrentLinkedQueue<>();
        this.mTasks = concurrentLinkedQueue;
        this.mContext = context;
        concurrentLinkedQueue.add(new CleanUpTask());
        executeTask(0L);
    }

    private void cleanExpiredTask() {
        while (!this.mTasks.isEmpty()) {
            Task taskPeek = this.mTasks.peek();
            if (taskPeek != null) {
                if (!taskPeek.isExpired() && this.mTasks.size() <= 6) {
                    return;
                }
                MyLog.v("remove Expired task");
                this.mTasks.remove(taskPeek);
            }
        }
    }

    /* JADX INFO: Access modifiers changed from: private */
    public void cleanUp() {
        if (SDCardUtils.isSDCardBusy() || SDCardUtils.isSDCardUnavailable()) {
            return;
        }
        try {
            File file = new File(this.mContext.getExternalFilesDir(null) + ZIPPED_LOG_PATH);
            if (file.exists() && file.isDirectory()) {
                for (File file2 : file.listFiles()) {
                    file2.delete();
                }
            }
        } catch (NullPointerException e) {
        }
    }

    private void executeTask(long j) {
        if (this.mTasks.isEmpty()) {
            return;
        }
        TaskExecutor.execute(new SerializedAsyncTaskProcessor.SerializedAsyncTask() { // from class: com.xiaomi.push.log.LogUploader.2
            SerializedAsyncTaskProcessor.SerializedAsyncTask current;

            @Override // com.xiaomi.channel.commonutils.misc.SerializedAsyncTaskProcessor.SerializedAsyncTask
            public void postProcess() {
                SerializedAsyncTaskProcessor.SerializedAsyncTask serializedAsyncTask = this.current;
                if (serializedAsyncTask != null) {
                    serializedAsyncTask.postProcess();
                }
            }

            @Override // com.xiaomi.channel.commonutils.misc.SerializedAsyncTaskProcessor.SerializedAsyncTask
            public void process() {
                Task task = (Task) LogUploader.this.mTasks.peek();
                if (task == null || !task.canExcuteNow()) {
                    return;
                }
                if (LogUploader.this.mTasks.remove(task)) {
                    this.current = task;
                }
                SerializedAsyncTaskProcessor.SerializedAsyncTask serializedAsyncTask = this.current;
                if (serializedAsyncTask != null) {
                    serializedAsyncTask.process();
                }
            }
        }, j);
    }

    public static LogUploader getInstance(Context context) {
        if (sInstance == null) {
            synchronized (LogUploader.class) {
                try {
                    if (sInstance == null) {
                        sInstance = new LogUploader(context);
                    }
                } catch (Throwable th) {
                    throw th;
                }
            }
        }
        sInstance.mContext = context;
        return sInstance;
    }

    /* JADX INFO: Access modifiers changed from: private */
    public void uploadIfNeed(long j) {
        Task taskPeek = this.mTasks.peek();
        if (taskPeek == null || !taskPeek.canExcuteNow()) {
            return;
        }
        executeTask(j);
    }

    public void checkUpload() {
        cleanExpiredTask();
        uploadIfNeed(0L);
    }

    public void upload(final String str, final String str2, final Date date, final Date date2, final int i, final boolean z) {
        this.mTasks.add(new Task() { // from class: com.xiaomi.push.log.LogUploader.1
            File file;

            @Override // com.xiaomi.channel.commonutils.misc.SerializedAsyncTaskProcessor.SerializedAsyncTask
            public void postProcess() {
                File file = this.file;
                if (file != null && file.exists()) {
                    LogUploader.this.mTasks.add(LogUploader.this.new UploadTask(str, str2, this.file, z));
                }
                LogUploader.this.uploadIfNeed(0L);
            }

            @Override // com.xiaomi.push.log.LogUploader.Task, com.xiaomi.channel.commonutils.misc.SerializedAsyncTaskProcessor.SerializedAsyncTask
            public void process() {
                if (SDCardUtils.isSDCardUseful()) {
                    try {
                        File file = new File(LogUploader.this.mContext.getExternalFilesDir(null) + LogUploader.ZIPPED_LOG_PATH);
                        file.mkdirs();
                        if (file.isDirectory()) {
                            LogFilter logFilter = new LogFilter();
                            logFilter.setMaxLen(i);
                            this.file = logFilter.filter(LogUploader.this.mContext, date, date2, file);
                        }
                    } catch (NullPointerException e) {
                    }
                }
            }
        });
        executeTask(0L);
    }
}

File: legacy-runtime/src/main/java/com/xiaomi/push/log/LogUploader.kt
package com.xiaomi.push.log

import android.content.Context
import android.content.SharedPreferences
import com.xiaomi.channel.commonutils.file.SDCardUtils
import com.xiaomi.channel.commonutils.logger.LoggerInterface
import com.xiaomi.channel.commonutils.logger.MyLog
import com.xiaomi.channel.commonutils.misc.SerializedAsyncTaskProcessor
import com.xiaomi.channel.commonutils.network.Network
import com.xiaomi.miui.pushads.sdk.NotifyAdsDef
import com.xiaomi.push.service.PushConstants
import com.xiaomi.push.service.ServiceConfig
import com.xiaomi.smack.util.TaskExecutor
import org.json.JSONException
import org.json.JSONObject
import java.io.File
import java.util.*
import java.util.concurrent.ConcurrentLinkedQueue

class LogUploader private constructor(context: Context) {
    private val mTasks: ConcurrentLinkedQueue<Task> = ConcurrentLinkedQueue()
    private var mContext: Context = context

    open inner class Task : SerializedAsyncTaskProcessor.SerializedAsyncTask() {
        var timestamp: Long = System.currentTimeMillis()

        open fun canExcuteNow(): Boolean = true

        internal fun isExpired(): Boolean = System.currentTimeMillis() - timestamp > EXPIRE_TIME

        override fun process() {}
    }

    inner class CleanUpTask : Task() {
        override fun process() = cleanUp()
    }

    inner class UploadTask(
        val url: String,
        val token: String,
        val file: File,
        val force: Boolean
    ) : Task() {
        var retryNum: Int = 0
        var uploaded: Boolean = false

        @Throws(JSONException::class)
        private fun checkLimit(): Boolean {
            val prefs = mContext.getSharedPreferences(PREF_NAME, 0)
            var jsonString = prefs.getString(PREF_KEY_REQUEST, "")
            var currentTime = System.currentTimeMillis()
            var times = 0
            try {
                val json = JSONObject(jsonString)
                currentTime = json.getLong(NotifyAdsDef.JSON_TAG_ACTIONTIME)
                times = json.getInt("times")
            } catch (e: JSONException) {
            }
            var newTimes = times
            if (System.currentTimeMillis() - currentTime < 86400000) {
                if (times > MAX_TIMES_PER_DAY) return false
            } else {
                currentTime = System.currentTimeMillis()
                newTimes = 0
            }
            return try {
                val json = JSONObject()
                json.put(NotifyAdsDef.JSON_TAG_ACTIONTIME, currentTime)
                json.put("times", newTimes + 1)
                prefs.edit().putString(PREF_KEY_REQUEST, json.toString()).commit()
                true
            } catch (e: JSONException) {
                MyLog.v("JSONException on put " + e.message)
                true
            }
        }

        override fun canExcuteNow(): Boolean {
            return Network.isWIFIConnected(mContext) || (force && Network.hasNetwork(mContext))
        }

        override fun postProcess() {
            if (!uploaded) {
                retryNum++
                if (retryNum < 3) {
                    mTasks.add(this@UploadTask)
                }
            }
            if (uploaded || retryNum >= 3) {
                file.delete()
            }
            this@LogUploader.uploadIfNeed((1 shl retryNum).toLong() * 1000)
        }

        override fun process() {
            try {
                if (checkLimit()) {
                    val map: MutableMap<String, String> = mutableMapOf()
                    map["uid"] = ServiceConfig.getDeviceUUID()!!
                    map["token"] = token
                    map["net"] = Network.getActiveConnPoint(mContext)!!
                    Network.uploadFile(url, map as Map<String, String>, file, PushConstants.UPLOAD_FILE_POST_KEY)
                }
                uploaded = true
            } catch (e: Exception) {
            }
        }
    }

    private fun cleanExpiredTask() {
        while (mTasks.isNotEmpty()) {
            val task = mTasks.peek() ?: break
            if (!task.isExpired() && mTasks.size <= MAX_PENDING_TASKS) return
            MyLog.v("remove Expired task")
            mTasks.remove(task)
        }
    }

    private fun cleanUp() {
        if (SDCardUtils.isSDCardBusy() || SDCardUtils.isSDCardUnavailable()) return
        try {
            val dir = File(mContext.getExternalFilesDir(null).toString() + ZIPPED_LOG_PATH)
            if (dir.exists() && dir.isDirectory) {
                dir.listFiles()?.forEach { it.delete() }
            }
        } catch (e: NullPointerException) {
        }
    }

    private fun executeTask(delay: Long) {
        if (mTasks.isEmpty()) return
        TaskExecutor.execute(object : SerializedAsyncTaskProcessor.SerializedAsyncTask() {
            var current: SerializedAsyncTaskProcessor.SerializedAsyncTask? = null
            override fun postProcess() {
                current?.postProcess()
            }
            override fun process() {
                val task = mTasks.peek() as Task?
                if (task == null || !task.canExcuteNow()) return
                if (mTasks.remove(task)) {
                    current = task
                }
                current?.process()
            }
        }, delay)
    }

    private fun uploadIfNeed(delay: Long) {
        val task = mTasks.peek() as Task?
        if (task == null || !task.canExcuteNow()) return
        executeTask(delay)
    }

    fun checkUpload() {
        cleanExpiredTask()
        uploadIfNeed(0L)
    }

    fun upload(url: String, token: String, startDate: Date, endDate: Date, maxLen: Int, force: Boolean) {
        val task = object : Task() {
            var file: File? = null
            override fun postProcess() {
                if (file != null && file!!.exists()) {
                    mTasks.add(UploadTask(url, token, file!!, force))
                }
                this@LogUploader.uploadIfNeed(0L)
            }
            override fun process() {
                if (SDCardUtils.isSDCardUseful()) {
                    try {
                        val dir = File(mContext.getExternalFilesDir(null).toString() + ZIPPED_LOG_PATH)
                        dir.mkdirs()
                        if (dir.isDirectory) {
                            val filter = LogFilter()
                            filter.setMaxLen(maxLen)
                            file = filter.filter(mContext, startDate, endDate, dir)
                        }
                    } catch (e: NullPointerException) {
                    }
                }
            }
        }
        mTasks.add(task)
        executeTask(0L)
    }

    companion object {
        private const val EXPIRE_TIME = 172800000L
        private const val MAX_PENDING_TASKS = 6
        private const val MAX_TIMES_PER_DAY = 10
        private const val PREF_KEY_REQUEST = "log.requst"
        private const val PREF_NAME = "log.timestamp"
        private const val ZIPPED_LOG_PATH = "/.logcache"

        @Volatile
        private var sInstance: LogUploader? = null

        fun getInstance(context: Context): LogUploader {
            if (sInstance == null) {
                synchronized(LogUploader::class.java) {
                    if (sInstance == null) {
                        sInstance = LogUploader(context)
                    }
                }
            }
            sInstance!!.mContext = context
            return sInstance!!
        }
    }

    init {
        mTasks.add(CleanUpTask())
        executeTask(0L)
    }
}
